wire/error.go

```diff
@@ -9,6 +9,176 @@ import (
 	"fmt"
 )
 
+// ErrorCode describes a kind of message error.
+type ErrorCode int
+
+// These constants are used to identify a specific Error.
+const (
+	// ErrNonCanonicalVarInt is returned when a variable length integer
+	// is encoded non-canonically.
+	ErrNonCanonicalVarInt ErrorCode = iota + 1
+
+	// ErrVarStringTooLong is returned when a variable string exceeds the
+	// maximum message size allowed.
+	ErrVarStringTooLong
+
+	// ErrVarBytesTooLong is returned when a variable byte slice exceeds the
+	// maximum message size allowed.
+	ErrVarBytesTooLong
+
+	// ErrCmdTooLong is returned when a command exceeds the maximum command
+	// size allowed.
+	ErrCmdTooLong
+
+	// ErrPayloadTooLarge is returned when a payload exceeds the maximum
+	// payload size allowed.
+	ErrPayloadTooLarge
+
+	// ErrWrongNetwork is returned when a message intended for a different
+	// network is received.
+	ErrWrongNetwork
+
+	// ErrMalformedCmd is returned when a malformed command is received.
+	ErrMalformedCmd
+
+	// ErrUnknownCmd is returned when an unknown command is received.
+	ErrUnknownCmd
+
+	// ErrPayloadChecksum is returned when a message with an invalid checksum
+	// is received.
+	ErrPayloadChecksum
+
+	// ErrTooManyAddrs is returned when an address list exceeds the maximum
+	// allowed.
+	ErrTooManyAddrs
+
+	// ErrTooManyTxs is returned when a the number of transactions exceed the
+	// maximum allowed.
+	ErrTooManyTxs
+
+	// ErrInvalidVersion is returned when an invalid message version is
+	// received.
+	ErrInvalidVersion
+
+	// ErrFilterTooLarge is returned when a committed filter exceeds
+	// the maximum size allowed.
+	ErrFilterTooLarge
+
+	// ErrTooManyProofs is returned when the numeber of proof hashes
+	// exceeds the maximum allowed.
+	ErrTooManyProofs
+
+	// ErrTooManyFilterTypes is returned when the number of filter types
+	// exceeds the maximum allowed.
+	ErrTooManyFilterTypes
+
+	// ErrTooManyLocators is returned when the number of block locators exceed
+	// the maximum allowed.
+	ErrTooManyLocators
+
+	// ErrTooManyVectors is returned when the number of inventory vectors
+	// exceed the maximum allowed.
+	ErrTooManyVectors
+
+	// ErrTooManyHeaders is returned when the number of block headers exceed
+	// the maximum allowed.
+	ErrTooManyHeaders
+
+	// ErrHeaderContainsTxs is returned when a header's transactions
+	// count is greater than zero.
+	ErrHeaderContainsTxs
+
+	// ErrTooManyVotes is returned when the number of vote hashes exceed the
+	// maximum allowed.
+	ErrTooManyVotes
+
+	// ErrTooManyBlocks is returned when the number of block hashes exceed the
+	// maximum allowed.
+	ErrTooManyBlocks
+
+	// ErrUnequalQuantities is returned when transactions quantities expected
+	// to be equal are not.
+	ErrUnequalQuantities
+
+	// ErrUnknownTxType is returned when a transaction type is unknown.
+	ErrUnknownTxType
+
+	// ErrWitnessOnlyTx is returned when reading the prefix for a witness only
```

Comment:
Not very descriptive.  Without me going to look under what circumstances the error is returned, this is completely ambiguous to me.